<?xml version="1.0"?>
<launch>

	<param name="use_sim_time" value="true"/>

    <!--Trackdrive Nodes-->
	<node name="path_planner" pkg="fs_planning-2022" type="path_planner.py" output="screen"/>
	<!--<node name="path_planner_wv" pkg="fs_planning" type="path_planner_wv.py" output="screen"/>-->
	<!--<node name="path_planner_planC" pkg="fs_planning" type="path_planner_planC.py" output="screen"/>-->

	<!--Skidpad Nodes-->
	<!--<node name="path_planner_skidpad1" pkg="fs_planning" type="path_planner_skidpad1.py" output="screen"/>-->
	<!--<node name="path_planner_skidpad2" pkg="fs_planning" type="path_planner_skidpad2.py" output="screen"/>-->
	<!--<node name="path_planner_skidpad3" pkg="fs_planning" type="path_planner_skidpad3.py" output="screen"/>-->

    <!--Plotting-->
	<param name="planner/real_time_local_plotting" value="True"/>
    <param name="planner/real_time_global_plotting" value="True"/>
    <param name="planner/all_waypoints_plotting" value="True"/>

    <!--Cones Styles-->
    <param name="planner/big_cone_style" value="3"/>
    <param name="planner/orange_cone_style" value="2"/>
    <param name="planner/yellow_cone_style" value="1"/>
    <param name="planner/blue_cone_style" value="0"/>

    <!--Parameters-->
    <param name="planner/number_of_waypoints_needed" value="4"/>
    <param name="planner/max_distance_of_view" value="10"/>
    <param name="planner/track_width" value="3"/>
    <param name="planner/all_waypoints_filtering_threshold" value="6"/> 

    <!--Topics-->
    <param name="planner/map_topic" value="airsim/Cones"/> 
    <param name="planner/pose_topic" value="airsim/KinematicOdometry"/>
    <param name="planner/waypoints_topic" value="fast_slam2/ConeGlobalMap"/> 

    <!--Costs Weights Trackdrive-->
    <param name="/planner/distance_weight" value="2"/>
    <param name="/planner/angle_weight" value="1"/>
    <param name="/planner/voronoi_weight" value="1"/>

    <!--Costs Weights Skidpad-->
    <param name="/planner/distance_weight_skid" value="1"/>
    <param name="/planner/angle_weight_skid" value="200"/>
    <param name="/planner/voronoi_weight_skid" value="1"/>
    <param name="/planner/position_weight_skid" value="1"/>
    <param name="/planner/left_right_cones_weight_skid" value="1"/>
    
	
</launch>

